### ISSUE TITLE
Sequational OAuth authentication

### ISSUE THREAD

**Issue by sobolevn** [axnsan12/drf-yasg#169]
Hi, thanks for this tool.

I am using two OAuth providers in my project. The first one (`gitlab`) is just a basic OAuth provider. It works perfectly fine.

But, I also have another provider (`wakatime`). As I call it: a service provider. It works like "connect accounts". And that why it needs current `request.user` to connect the account to the existing user.

That's what it looks like for now:

```python
_swagger_redirect = '/static/drf-yasg/swagger-ui-dist/oauth2-redirect.html'

SWAGGER_SETTINGS = {
    # 'USE_SESSION_AUTH': False,
    'OAUTH2_REDIRECT_URL': config('BASE_URL') + _swagger_redirect,
    'SECURITY_DEFINITIONS': {
        'gitlab': {
            'type': 'oauth2',
            'authorizationUrl': 'https://gitlab.com/oauth/authorize',
            'tokenUrl': config('API_URL') + '/auth/gitlab/',
            'flow': 'accessCode'
        },
        'wakatime': {  # TODO: change Wakatime app and settings
            'type': 'oauth2',
            'authorizationUrl': 'https://wakatime.com/oauth/authorize',
            'tokenUrl': config('API_URL') + '/auth/wakatime/',
            'flow': 'accessCode',
            'scopes': {
                'email,read_logged_time': 'default required Wakatime scope',
            },
        },
    },
    'SECURITY_REQUIREMENTS': [
        {'gitlab': [], 'wakatime': []},
    ],
}
```

And, as I said, `gitlab` provider looks and works perfectly fine.
<img width="647" alt="2018-07-24 10 39 38" src="https://user-images.githubusercontent.com/4660275/43123889-38353a88-8f2e-11e8-9e94-8999f494b3a3.png">

`wakatime` provider also works, it authenticates agains real `wakatime` OAuth, but then my server fails. Because existing auth header is not sent to my server. So, `request.user.is_authenticated` fails.

<img width="543" alt="2018-07-24 10 43 23" src="https://user-images.githubusercontent.com/4660275/43124019-8def30c8-8f2e-11e8-874d-935058ab9681.png">
<img width="1257" alt="2018-07-24 10 42 32" src="https://user-images.githubusercontent.com/4660275/43124020-8e0db228-8f2e-11e8-90d1-a8f50f4991bf.png">

Do you have any specific solution in mind? Thanks!

**Comment by axnsan12** [comment]
If I'm getting this right then this is the same as https://github.com/axnsan12/drf-yasg/issues/131? In that case I have nothing for you, it's a bug that has to be fixed in swagger-api/swagger-ui#4334, which seems to have dried up a bit...

**Comment by axnsan12** [comment]
Closing this for lack of information. Feel free to reopen with more details.
